Gather the mean values of T2 from every BA in the Historical dataset and create a time series analysis

In [1]:
import os

import pandas as pd

import numpy as np

import matplotlib

import glob

In [2]:
# set up a path for the concatenated dataframe
from pathlib import Path

path = r'C:\projects\CONUS_TGW_WRF_Historical\CONUS_TGW_WRF_Historical'

files = Path(path).glob('*.csv')

In [3]:
# create a concatenated dataframe from all csv files
dfs = list()
for f in files:
    data = pd.read_csv(f)
    # .stem is method for pathlib objects to get the filename w/o the extension
    data['file'] = f.stem
    dfs.append(data)

In [4]:
# concatenate big dataframe of all data
plant_df = pd.concat(dfs, ignore_index=True)

In [5]:
# create a list of the data frames
path = r'C:\projects\CONUS_TGW_WRF_Historical\CONUS_TGW_WRF_Historical'
all_files = glob.glob(os.path.join(path , "*.csv"))

li = []

for filename in all_files:
    df = pd.read_csv(filename, index_col=None, header=0)
    li.append(df)

In [13]:
# return a list of the means of T2 for each imported dataframe
mean_list = []
for df in li:
    mean_list.append(df["T2"].mean())
    
mean_df = pd.DataFrame(mean_list, columns=["T2 Means"])

        T2 Means
0     286.316761
1     286.469531
2     285.469139
3     285.432616
4     285.600995
...          ...
2275  281.200692
2276  281.189107
2277  280.206579
2278  279.044531
2279  278.161652

[2280 rows x 1 columns]


I'd like to get some more info into this mean_df dataframe. Ideally the name of the csv file so that I have the year and BA for each mean.

In [12]:
# run a for loop using basename function to get base file names into a list
file_names = []
for file_path in all_files:
    x = os.path.basename(file_path)
    file_names.append(x)

In [14]:
# add file_names column to mean_df to get T2 means and their respective file
mean_df['File'] = file_names

print(mean_df)

        T2 Means                                       File
0     286.316761  AECI_WRF_Hourly_Mean_Meteorology_1980.csv
1     286.469531  AECI_WRF_Hourly_Mean_Meteorology_1981.csv
2     285.469139  AECI_WRF_Hourly_Mean_Meteorology_1982.csv
3     285.432616  AECI_WRF_Hourly_Mean_Meteorology_1983.csv
4     285.600995  AECI_WRF_Hourly_Mean_Meteorology_1984.csv
...          ...                                        ...
2275  281.200692  WAUW_WRF_Hourly_Mean_Meteorology_2015.csv
2276  281.189107  WAUW_WRF_Hourly_Mean_Meteorology_2016.csv
2277  280.206579  WAUW_WRF_Hourly_Mean_Meteorology_2017.csv
2278  279.044531  WAUW_WRF_Hourly_Mean_Meteorology_2018.csv
2279  278.161652  WAUW_WRF_Hourly_Mean_Meteorology_2019.csv

[2280 rows x 2 columns]


Okay let's think this through. I want to end up with the means of T2 for each BA, but do I want the mean of each year, or the overall mean? Say we want the mean of each year, that's 2,280 means. Which is probably easier to achieve and potentially better...So we need a mean of T2 from each file, as each file represents one year for a particular BA. So in the directory, we can extract each file. Then for that file, calculate the mean of the T2 column. Then input that mean into a dataframe where one axis is the BA, and the other axis is the year. Ideally. 